sdk/cosmos/sdk/src/main/java/com/azure/data/cosmos/CosmosClientBuilder.java

```diff
@@ -186,6 +187,20 @@ public CosmosClient build() {
         return new CosmosClient(this);
     }
 
+    /**
+     * Builds a cosmos sync client object with the provided properties
+     * @return CosmosSyncClient
+     */
+    public CosmosSyncClient buildSyncClient() {
+
+        ifThrowIllegalArgException(this.serviceEndpoint == null, "cannot build client without service endpoint");
+        ifThrowIllegalArgException(
+                this.keyOrResourceToken == null && (permissions == null || permissions.isEmpty()),
+                "cannot build client without key or resource token");
```

Comment:
update this check with `|| cosmosKeyCredential == null `

Reply:
extracted the validation into a method to handle this